src/parallel/KDDecomposition.cpp

```diff
@@ -864,7 +882,7 @@ bool KDDecomposition::decompose(KDNode* fatherNode, KDNode*& ownArea, MPI_Comm c
 		MPI_CHECK( MPI_Group_incl(origGroup, newNumProcs, &origRanks[0], &newGroup) );//create new MPI group based on rank (as calculated before)
 		MPI_CHECK( MPI_Comm_create(commGroup, newGroup, &newComm) );
 
-		KDNode* newOwnArea = NULL;
+		KDNode* newOwnArea = nullptr;
```

Comment:
really not a friend of
 a) raw pointers
 b) initializing with a `nullptr`
 c) not checking if the pointer actually was set in the end (probably...)

But since I guess this is used rampant throughout this code 🤷‍♂️

Reply:
welcome to the beauty of ls1 mardyn :P